security-analysis-api/src/main/java/com/powsybl/security/execution/SecurityAnalysisExecutionImpl.java

```diff
@@ -22,15 +23,25 @@
  */
 public class SecurityAnalysisExecutionImpl implements SecurityAnalysisExecution {
 
-    private final SecurityAnalysisFactory factory;
+    private final String providerName;
```

Comment:
A detail but I would be more in favor of having a `Runner` here :
 * it will allow to make the runner resolution only once and not for every execution
 * it is slightly more flexible since it can allow a third party user to provide its own runner (think for instance to wrap it in a runner which add some additional behaviour such as metrics/logging)

Reply:
`String providerName` was replaced by `SecurityAnalysis.Runner runner`